File: 一帆视频/js/jiami.js

// n ? (e = e.substring(1)) + (o ? "&" : "") + "vv=" + m.Md5.hashStr(s) + "&pub=" + t.publicKey : e + (o ? "&" : "?") + "vv=" + m.Md5.hashStr(s) + "&pub=" + t.publicKey
const { Md5 } = require('ts-md5');

// var privateKey;
// publicKeyInedx = (other Date).getTime();
// publicKey =  "CJSpCZWqDpauC2uvCJbVLLDVCpWkDpCkCZCuBZLVOJ9bEJ1YCp0uOp0nD3bYOZaoCpXaCsCuCZKmE3WrC6HVD3XYP3DXP3OpP3CrPZ8sCpKqEMKsE6HbCM4qCcKsOJ5"
// s = publicKey + "&" + '' + "&" + privateKey[publicKeyInedx % privateKey.length]
// var vv = m.Md5.hashStr(s)
// t.hashStr
// var pub = publicKey

s = 'CJSpCZWrEJ4rDIurD3HVLLDVCpWkDpCkCZCuBZLVOpSqD3WqCc5YD6CrD6LaDZbXCZasCMPZC30oCcDaCs9VDpOtEMPZCJWqCJSqCJKpDpGvPJPXPZPXOs4qDsOrP65&cinema=1&id=cwy9wuxg4zf&a=1&lang=none&usersign=1&region=us&device=1&ismastersupport=1&SpCZJSpCZWrEJ4rDIurD'

const hashValue = Md5.hashStr(s);

console.log(s); // 输出加密后的MD5值
console.log(hashValue); // 输出加密后的MD5值

File: 一帆视频/jiami.py
import time
import hashlib

def get_sign(publicKey,privateKey,flag=True,params=None):

    if flag:
        mid = ''
        public_key_index = int(time.time() * 1000)
    else:
        mid = params
        public_key_index = 0

    # 计算私钥索引
    private_key_index = public_key_index % len(privateKey)
    # 构建s字符串
    s = publicKey + "&" + mid + "&" + privateKey[private_key_index]
    print(s)
    md5 = hashlib.md5()
    md5.update(s.encode('utf-8'))
    return md5.hexdigest()


'CJSpCZWrEJ4rDIurD3HVLLDVCpWkDpCkCZCuBZLVOpSqD3WqCc5YD6CrD6LaDZbXCZasCMPZC30oCcDaCs9VDpOtEMPZCJWqCJSqCJKpDpGvPJPXPZPXOs4qDsOrP65&cinema=1&id=cwy9wuxg4zf&a=1&lang=none&usersign=1&region=us&device=1&ismastersupport=1&SpCZJSpCZWrEJ4rDIurD'
'CJSpCZWsC3OuDYuuC3XVLLDVCpWkDpCkCZCuBZLVE3HZP39aDcGmCJOoD6LYCs5XCMOoOs9bDc9cPZ0oP6LVPJaqE65aE34rOs9cDZ4uEJ5aDZ9aDMKrDJOnE3WuOs1&cinema=1&id=Cwy9wuxg4ZF&a=1&lang=none&usersign=1&region=US&device=1&isMasterSupport=1&SpCZJSpCZWsC3OuDYuuC'